### PR TITLE
Km.demo 404 update graph with state

### PR DESCRIPTION
**Pull request by kiley-mitti**
[DEMO-404](https://rocketcom.atlassian.net/browse/DEMO-404)
## Brief Description

This PR encompasses many changes to the track file data utility. The chart is populated with largely fake data. The delete/undo/filter work again.

I've added faker data to initial orbit properties for Azimuth and Elevation data.

Added table service to both 'select files' and 'filter/edit data'.

I've hooked up a trackFileDataUtility service that holds the track files that are selected in 'Track Data > Select Files' and makes them available to the graph and the table in filter/edit data.

Oh! I split the track-data components into track-data-graph and track-data-table. It could probably use further refactoring.

I recently merged main into this branch and there weren't conflicts at the time so I'm pretty confident that while these changes are wide ranging they're not interfering.

BIG THANKS to Maria for all her help on the Chart. That thing is a beast.

## Issues and/or Limitations

This could probably use a refactor but at this point the PR was getting so out of hand that I wanted to merge a working base in.

I have a followup ticket to revamp the CSS to fix some of the visual glitches.

## Final Checklist

- [x] Works in Chrome
- [x] Works in Firefox
- [x] Works in Safari
- [x] Handles responsiveness as required
- [x] Dark theme styles are correct
- [x] Light theme styles are correct

### CODE REVIEW

File: src/app/main/utility-components/track-data-utility/track-data/track-data.component.html [hunk 1]
```diff
@@ -44,124 +44,25 @@
...
-      </div>
-    </div>
-    <footer>
-      <span>OD Quality Est: 82/100</span>
-      <span>Data Points: {{ dataPointLength }}</span>
-    </footer>
+    <fds-track-data-graph #trackDataGraph></fds-track-data-graph>
   </div>
 
+  <!--Table Component: ToDo split into two components-->
```
Comment (Paterma) [inline src/app/main/utility-components/track-data-utility/track-data/track-data.component.html:64]: Can be removed 😁

**Review by Paterma — COMMENTED** [review]
**Great** work Kiley, this was a _beast_!
A couple small things. When you un-check some files, go to the chart, then back to the files, the selected checkboxes aren't maintained- it defaults back to them all being checked. This could def be handled in a separate ticket since this is so large if that's something we want to fix. 

The other small things are css related- when you open a drawer, the buttons above get a little wonky. 
Graph related- we should probably have design decide if they want the graph labels below to be horizontal or at an angle, since the chart will 'jump' in size depending on what way the labels are if you're filtering/zooming/have less shown etc. (Can also be a separate ticket- just making note 😃)
The title is also there twice, a second below with an icon

Screenshot 2023-10-09 at 1 12 16 PM
.

File: src/app/main/utility-components/track-data-utility/track-data/track-data.component.html (hunk 1 above)
Comment (kiley-mitti) [reply src/app/main/utility-components/track-data-utility/track-data/track-data.component.html]: Yup! I pushed a small commit that does this. :)

**Comment by kiley-mitti** [comment]
@Paterma 

> A couple small things. When you un-check some files, go to the chart, then back to the files, the selected checkboxes aren't maintained- it defaults back to them all being checked. This could def be handled in a separate ticket since this is so large if that's something we want to fix.

Yeah, I noticed that. It has to do with where the track files are being called initially. I should move that to the track file data utility service and have every component pull the shared data from there.. but I would love this to be a follow up refactor because it's going to take some time.


>The other small things are css related- when you open a drawer, the buttons above get a little wonky.
>Graph related- we should probably have design decide if they want the graph labels below to be horizontal or at an angle, since the chart will 'jump' in size depending on what way the labels are if you're filtering/zooming/have less shown etc. (Can also be a separate ticket- just making note 😃)
>The title is also there twice, a second below with an icon

All yup! Thank you for listing them out! I've made a css cleanup branch off this one which is going to solve most of this funkiness. I have already fixed the drawer buttons. Good point about the title. It really SHOULD only be in one spot. Let me go refer to the files and I'll fix that in my css cleanup too.

As to the graph labels... yeah.. we should figure that out. I don't know how much of design's time we have. That might be a thing we have to decide together. let me think on it. This cartesian charting thing is a real brain buster. 😂

**Review by NicDominguez — APPROVED** [review]

File: src/app/main/utility-components/track-data-utility/track-files-data.service.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  constructor(){}
+
+  get() {
+    return this.checkedTrackFiles.asObservable()
+  }
+
+  set(newData: TrackFile[]): void{
+    this.checkedTrackFiles.next(newData)
+    console.log(newData, 'newData')
+    console.log(this.checkedTrackFiles)
```
[the remark this answers is not in the document]
Comment (kiley-mitti) [reply src/app/main/utility-components/track-data-utility/track-files-data.service.ts:25]: haha I cleaned that up in my next incoming PR. I'm going to merge main into that one real quick too

**Pull request merged**